### PR TITLE
Setup Java 17 when building repo in github CI workflow

### PR DESCRIPTION
**Pull request by teamurko**
## Summary
Update setup java action to build with Java17, update docs. The change should be non-consequential since the source and target byte code are still Java 1.8 compatible.

## Changes

- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [ ] Bug Fixes
- [ ] New Features
- [ ] Performance Improvements
- [ ] Code Style
- [ ] Refactoring
- [x] Documentation
- [ ] Tests
- [x] Upgrade

For all the boxes checked, please include additional details of the changes made in this pull request.  

## Testing Done

- [ ] Manually Tested on local docker setup. Please include commands ran, and their output.
- [ ] Added new tests for the changes made.
- [ ] Updated existing tests to reflect the changes made.
- [x] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

For all the boxes checked, include a detailed description of the testing done for the changes made in this pull request.

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

**Review by rohitkum2506 — APPROVED** [review]

**Review by autumnust — CHANGES_REQUESTED** [review]
shall we have both version in CI for the time being?

File: .github/workflows/build-tag-publish.yml [hunk 1]
```diff
@@ -18,11 +18,11 @@ jobs:
           ref: ${{ github.event.pull_request.merge_commit_sha }}
           fetch-depth: 0
 
-      - name: Set up JDK 1.8
+      - name: Set up JDK 17
         uses: actions/setup-java@v1
         with:
           distribution: 'microsoft'
-          java-version: '1.8'
+          java-version: '17'
```
Comment (abhisheknath2011) [inline .github/workflows/build-tag-publish.yml:25]: Is it possible to support both the version while upgrade is in progress?

**Comment by teamurko** [comment]
> shall we have both version in CI for the time being?

It's byte code 8, no need to publish two artifacts, when we build for 2 targets we can publish both

**Review by maluchari — APPROVED** [review]

**Review by autumnust — APPROVED** [review]

File: .github/workflows/build-tag-publish.yml (hunk 1 above)
Comment (abhisheknath2011) [reply .github/workflows/build-tag-publish.yml:25]: So this should not impact internal OH upgrade based on comments in this PR.

**Review by abhisheknath2011 — APPROVED** [review]

**Pull request merged**